ref(issue-stream): Simplify issue details tooltip loading and pull out common styles

malwilley opened the pull request:
Two goals of this PR are:

1. Pull out styles from `<StackTracePreview />` into `< GroupPreviewHovercard />` so we can use this for performance issue previews tooltips as well.
2. Simplify the way `<StackTracePreview />` loads data so we don't have to replicate that for other issue types. Right now the logic says: On hover, wait 100ms, then start fetching data. But don't show the tooltip loader until 400ms have passed. I removed these timers and put the fetch into the tooltip body component so it simply fetches data when the tooltip is shown. ~I can put the 400ms wait time back in if we think that's still necessary, I just kept it out since it seemed more consistent to me to always show the tooltip at the same time (instead of being delayed by how long the request takes).~

malwilley commented:
I added back the 400ms hovercard content delay (with [useDelayedLoadingState](https://github.com/getsentry/sentry/blob/ed0c9abf6b3308f2106a647c52dfebaa0c1c41e9/static/app/components/groupPreviewTooltip/utils.tsx#L7)) since I didn't want to change the behavior too much. Updated the PR description to reflect that.

scttcper commented on static/app/components/groupPreviewTooltip/groupPreviewHovercard.tsx:
```diff
@@ -0,0 +1,81 @@
+  const theme = useTheme();
+
+  // No need to preview on hover for small devices
+  const shouldNotPreview = useMedia(`(max-width: ${theme.breakpoints.large})`);
```
dang when did we get this

scttcper replied:
I think this should be medium
```suggestion
  const shouldNotPreview = useMedia(`(max-width: ${theme.breakpoints.medium})`);
```

malwilley replied:
The cutoff was large before, but I knocked it down to medium because I agree, I think that's better

scttcper commented:
I think this is triggering the request too early, we had added the delay on the request to avoid load on snuba if someone moves their mouse down the page. This seems to be triggering the request and cancelling if the cursor moves away?

scttcper commented:
i think it might be firing it at the same time, but i think cancelling it might be unnecessary? Not sure

malwilley commented on static/app/components/groupPreviewTooltip/groupPreviewHovercard.tsx:
```diff
@@ -0,0 +1,81 @@
+      visibility: hidden;
+    `}
+
+  ${Body} {
```
Done in fe59f2291011b459f75842c5990f7d6628914f45

Had to make a new component to forward the styled component class name, which I _think_ is how I need to do this? Either that or change `Hovercard` to accept something different than `bodyClassName`. I might looking into doing that later

malwilley commented:
> I think this is triggering the request too early, we had added the delay on the request to avoid load on snuba if someone moves their mouse down the page. This seems to be triggering the request and cancelling if the cursor moves away?

The request is firing when the hovercard mounts, which by default is 50ms. I changed to `delay={100}` (6a06f3bad518fc5e220b121d7c0a4edcebeda41e) to give it the same delay it had before in

scttcper reviewed (approved):
removing that component selector is 💫
